hypothesis-python/src/hypothesis/core.py

```diff
@@ -826,15 +829,20 @@ def execute_once(
             @proxies(self.test)
             def test(*args, **kwargs):
                 arg_drawtime = math.fsum(data.draw_times.values())
+                arg_gctime = gc_cumulative_time()
                 start = time.perf_counter()
                 try:
                     result = self.test(*args, **kwargs)
                 finally:
                     finish = time.perf_counter()
+                    in_gctime = gc_cumulative_time() - arg_gctime
                     in_drawtime = math.fsum(data.draw_times.values()) - arg_drawtime
-                    runtime = datetime.timedelta(seconds=finish - start - in_drawtime)
+                    runtime = datetime.timedelta(
+                        seconds=finish - start - in_drawtime - in_gctime
+                    )
                     self._timing_features = {
-                        "execute:test": finish - start - in_drawtime,
+                        "execute:test": runtime.total_seconds(),
+                        "overall:gc": in_gctime,
                         **data.draw_times,
                         **data._stateful_run_times,
```

Comment:
I don't think it's worth using a timedelta here - we're both starting and finishing with seconds.

The other problem is actually my fault, from https://github.com/HypothesisWorks/hypothesis/pull/3890/ - we're double-counting the `data._stateful_run_times`, and should subtract their sum from the `execute:test` span.  Mind handling that while you're here, and noting it in the changelog?